mono/metadata/boehm-gc.c

```diff
@@ -34,6 +34,11 @@
 #include <mono/utils/mono-os-mutex.h>
 #include <mono/utils/mono-counters.h>
 
+#ifdef _MSC_VER
+	// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+	void __mono_win32_boehm_gc_quiet_lnk4221(void) {}
```

Comment:
#pragma warning only works for compile warnings and not linker warnings, unfortunately. For compiler warnings it is also possible to disable warnings per source file (in the vcxproj file), but not supported for linker warnings. I agree that #pragma disable or per disable per source file in vcxproj would have been much clearer/better (that's one approach I usually resort to if there are no better options). So based on that we have a couple of options, leave as is not fixing the linker warnings, disable this warning for the complete libmonoruntime project, or keep proposed solution.